.github/workflows/build.yml

```diff
@@ -359,21 +359,28 @@ jobs:
 
   release:
     name: Release
-
     if: startsWith(github.ref, 'refs/tags/') || github.event_name == 'workflow_dispatch' && github.event.inputs.publish == 'true'
-
     runs-on: ubuntu-latest
-    environment: Publish
+    environment: pypipublish
     needs: [linux, windows, macos_x86, macos_aarch64, sdist]
+    permissions:
+      id-token: write
+      contents: read
     steps:
+      - uses: actions/checkout@v4
       - uses: actions/download-artifact@v4
         with:
           pattern: wheels-*
-          name: wheels
+          path: dist
+          merge-multiple: true
+      - name: List contents of dist directory
+        run: ls -la dist/
```

Comment:
probably don't need this now that it works, but I think I'd leave it in anyway, just to be sure

Reply:
Sounds good. We can remove it in subsequent PRs. And you're right, it's nice to have here to see what goes in.